zebra-chain/src/parameters/network/testnet.rs

```diff
@@ -144,6 +161,15 @@ impl ParametersBuilder {
         self
     }
 
+    /// Parses a hex-encoded block hash and sets the genesis hash to be used in the [`Parameters`] being built.
+    pub fn with_genesis_hash(mut self, genesis_hash: impl fmt::Display) -> Self {
+        self.genesis_hash = genesis_hash
+            .to_string()
+            .parse()
+            .expect("configured genesis hash must parse");
```

Comment:
This could be good cleanup, it's probably less important since the builder struct is only used at startup, so I'd prefer to defer it until after the PRs have merged, I opened #8432 so we don't forget. I think the intention of panicking in these methods was conciseness where the methods are called and to avoid duplicate code, but in hindsight, that doesn't seem important.

Reply:
> the intention of panicking in these methods was conciseness where the methods are called and to avoid duplicate code

This was my motivation the other day as well, and I also opened an issue to fix it.

Let's got with the current implementation.